railties/lib/rails/generators/rails/app/templates/Gemfile.tt

```diff
@@ -18,25 +20,24 @@ ruby <%= "'#{RUBY_VERSION}'" -%>
 <% end -%>
 <% end -%>
 
-# Use Active Model has_secure_password
-# gem 'bcrypt', '~> 3.1.7'
-<% unless skip_active_storage? -%>
 
-# Use Active Storage variant
-# gem 'image_processing', '~> 1.2'
-<% end -%>
+# Optional gems needed by specific Rails features:
 
-<% if depend_on_bootsnap? -%>
-# Reduces boot times through caching; required in config/boot.rb
-gem 'bootsnap', '>= 1.4.4', require: false
+# Use bcrypt to encrypt passwords securely. Works with https://guides.rubyonrails.org/active_model_basics.html#securepassword
+# gem 'bcrypt', '~> 3.1.7'
 
-<%- end -%>
+<% unless skip_action_cable? -%>
+# If using ActionCable, use the Redis adapter. https://guides.rubyonrails.org/action_cable_overview.html
+# gem 'redis', '~> 4.0'
+
+<% end -%>
 <%- if options.api? -%>
 # Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
 # gem 'rack-cors'
 
 <%- end -%>
 <% if RUBY_ENGINE == 'ruby' -%>
+# The gems below are used in development, but if they cause problems it's OK to remove them
```

Comment:
Personally, I'm not sure this line is necessary? Users new to rails, may already know about how gems work and I'm not sure rails should explain this?

(However, when I started using rails, I did have this very question, why is development used, it was not until I learnt about rubygems, then it all made sense).

Reply:
> (However, when I started using rails, I did have this very question, why is development used, it was not until I learnt about rubygems, then it all made sense).

I was in the same boat as you, which is why I added this line.